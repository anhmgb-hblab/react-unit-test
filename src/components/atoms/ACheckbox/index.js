import React from 'react';
import { Checkbox } from 'antd';
import PropTypes from 'prop-types';

const ACheckbox = ({
  checked,
  disabled,
  onChange,
  label,
}) => (
  <Checkbox
    checked={checked}
    disabled={disabled}
    onChange={onChange}
  >
    {label}
  </Checkbox>
);

ACheckbox.defaultProps = {
  checked: false,
  disabled: false,
  onChange: null,
  label: null,
};

ACheckbox.propTypes = {
  checked: PropTypes.bool,
  disabled: PropTypes.bool,
  onChange: PropTypes.func,
  label: PropTypes.string,
};

export default ACheckbox;
