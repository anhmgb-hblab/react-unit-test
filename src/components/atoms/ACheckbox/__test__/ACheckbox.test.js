import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import ACheckbox from '../index';

describe('Test ACheckbox running correctly', () => {
  it('test checkbox props', () => {
    const wrapper = mount(<ACheckbox label="Primary" />);
    expect(wrapper.props().checked).toEqual(false);
    expect(wrapper.props().disabled).toEqual(false);
    expect(wrapper.text()).toEqual('Primary');
  });

  it('test checkbox onChange', () => {
    const onChange = jest.fn(() => 'checked');
    const wrapper = mount(<ACheckbox onChange={onChange} />);
    wrapper.props().onChange();

    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange.mock.results[0].value).toBe('checked');
  });

  it('test rendering', () => {
    const tree = mount(<ACheckbox label="Primary" />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
