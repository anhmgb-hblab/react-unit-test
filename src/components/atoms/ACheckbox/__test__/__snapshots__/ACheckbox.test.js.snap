// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ACheckbox running correctly test rendering 1`] = `
<ACheckbox
  checked={false}
  disabled={false}
  label="Primary"
  onChange={null}
>
  <Checkbox
    checked={false}
    disabled={false}
    onChange={null}
  >
    <label
      className="ant-checkbox-wrapper"
    >
      <Checkbox
        checked={false}
        className=""
        defaultChecked={false}
        disabled={false}
        onBlur={[Function]}
        onChange={null}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        prefixCls="ant-checkbox"
        style={Object {}}
        type="checkbox"
      >
        <span
          className="ant-checkbox"
          style={Object {}}
        >
          <input
            checked={false}
            className="ant-checkbox-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            type="checkbox"
          />
          <span
            className="ant-checkbox-inner"
          />
        </span>
      </Checkbox>
      <span>
        Primary
      </span>
    </label>
  </Checkbox>
</ACheckbox>
`;
